desc/compute/_curve.py

```diff
@@ -639,6 +639,88 @@ def _x_sss_FourierXYZCurve(params, transforms, profiles, data, **kwargs):
     return data
 
 
+def _splinexyz_helper(f, p_knots, transforms, xq, kwargs, derivative):
+    """Used to compute XYZ coordinates for the SplineXYZCurve compute functions.
+
+    Parameters
+    ----------
+    f : list of ndarray
+        X, Y, Z function coords with shape (3, len(p_knots))
+    p_knots : ndarray
+        knots that come from params
+    transforms : dict
+        the transforms from the compute function
+    xq : ndarray
+        query points that come from s parameterization
+    kwargs : dict
+        the kwargs from the compute function
+    derivative : int
+        derivative order used for interpolation
+
+    Returns
+    -------
+    coords : ndarray
+        Interpolated XYZ coords with shape (3, len(xq))
+    """
+    method = kwargs.get("method", "cubic")
+    transforms["intervals"] = jnp.asarray(transforms["intervals"])
+    has_break_points = len(transforms["intervals"][0])
+    n_dim = 3
+
+    def inner_body(f, knots, period=None):
+        """Interpolation for spline curves."""
+        fq = [
+            interp1d(
+                xq,
+                knots,
+                f[i],
```

Comment:
interpax should be able to handle interpolating multiple funcitons at the same time, so you can pass in f as a nx3 array here instead of looping

Reply:
just passing in `f` isn't working for me, it gives me this error: `ValueError: x and f must be arrays of equal length`

Reply:
i think you just need to transpose it (and maybe also the output).

`f` is shape (3, N), interpax expects it to be (N,3) and will return something (M,3)

Reply:
Ah my bad, thanks!